app/cdap/components/Metadata/SearchBar/index.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState } from 'react';
+import T from 'i18n-react';
+import Search from '@material-ui/icons/Search';
+import styled from 'styled-components';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import Paper from '@material-ui/core/Paper';
+import InputBase from '@material-ui/core/InputBase';
+import IconButton from '@material-ui/core/IconButton';
+import ResourceCenterButton from 'components/ResourceCenterButton';
+
+const I18N_PREFIX = 'features.MetadataSearch';
+
+const Container = styled.div`
+  background: #dfdfdf;
+  border-bottom: 1px solid #cdcdcd;
+  padding: 10px;
+`;
+
+const SearchForm = styled(Paper)`
+  display: flex;
+  align-items: center;
+  padding: 2px 4px;
+  min-width: 400px;
+  width: 25%;
+  height: 40px;
+`;
+
+const SearchInput = styled(InputBase)`
+  flex: 1;
+  margin-left: 8px;
+  font-size: 13px;
+`;
+
+const ResourceAddButton = styled(ResourceCenterButton)`
+  top: 80px;
+`;
+
+interface ISearchBar {
+  query: string;
+}
+
+const SearchBar: React.FC<ISearchBar> = ({ query }) => {
+  const searchQueryRegex = new RegExp('^(?![*]).*$');
+  const searchPlaceholder = T.translate(`${I18N_PREFIX}.searchPlaceholder`);
+  const [error, setError] = useState(false);
+
+  const handleSearch = (event) => {
+    const searchQuery = event.target.value;
+    const isValid = searchQueryRegex.test(searchQuery);
+    setError(!isValid);
+    if (event.key === 'Enter' && searchQuery.trim() !== '' && isValid) {
+      window.location.href = `/ns/${getCurrentNamespace()}/metadata/search/${searchQuery.trim()}/result`;
+    }
+  };
+
+  return (
+    <>
+      <Container>
+        <SearchForm>
+          <SearchInput
```

Comment:
Maybe not this PR, but we should standardize our search input look and behavior. See also the one in the connection browser.

Reply:
https://cdap.atlassian.net/browse/CDAP-18520